flink/v1.19/flink/src/main/java/org/apache/iceberg/flink/TableLoader.java

```diff
@@ -54,7 +54,7 @@ static TableLoader fromHadoopTable(String location) {
     return fromHadoopTable(location, FlinkCatalogFactory.clusterHadoopConf());
   }
 
-  static TableLoader fromHadoopTable(String location, Configuration hadoopConf) {
+  static TableLoader fromHadoopTable(String location, Object hadoopConf) {
```

Comment:
I hate this part of the change.
We have a public API where type is not defined.
Do we have any better solution for this?

Reply:
What do you think of the `ParquetConfiguration` that Parquet-Java did: https://github.com/apache/parquet-java/pull/1141/ ?

Reply:
it seems quite a bit of effort to define a `IcebergHadoopConfiguration` class like parquet-java.

another possible option is to add a new overloaded method with Flink `Configuration` arg. Flink provides a util method for the conversion.
```
public class HadoopUtils {
    @SuppressWarnings("deprecation")
    public static Configuration getHadoopConfiguration(
            org.apache.flink.configuration.Configuration flinkConfiguration) 
```